src/server/common.h

```diff
@@ -117,6 +118,33 @@ struct OpArgs {
   }
 };
 
+// A strong type for a lock tag. Helps to disambiguide between keys and the parts of the
+// keys that are used for locking.
+class LockTag {
+  std::string_view str_;
+
+ public:
+  using is_stackonly = void;  // marks that this object does not use heap.
+
+  LockTag() = default;
+  explicit LockTag(std::string_view key);
+
+  explicit operator std::string_view() const {
+    return str_;
+  }
+
+  LockFp Fingerprint() const;
```

Comment:
Do you plan to cache this in the future?

Reply:
yes, in the next PR.